src/tests/ftest/util/apricot/apricot/test.py

```diff
@@ -550,6 +550,7 @@ def __init__(self, *args, **kwargs):
         # self.debug = False
         # self.config = None
         self.job_manager = None
+        self.pool_labels = []
```

Comment:
Use a single object to create unique pool labels - see other comments for details:
```suggestion
        self.label_generator = LabelGenerator()
```